### ISSUE TITLE
Unknown/Undefined behavior with Patient

### ISSUE THREAD

**Issue by jackcmeyer** [cqframework/clinical_quality_language#332]
There appears to be some unknown/undefined behavior around Patient .

Using the following CQL: 

```
library test version '0.0.000'

using QDM version '5.3'

context Patient

define "Initial Population":
	"Patient"

define "Patient":
	true
```

The result type of my definition `Initial Population` is `QDM.Patient`. Also, the result type of the definition `Patient` is `QDM.Patient`, but is pointing to the definition defined at `10:1-11:5`. This definition should return a boolean, however it appears to be referrencing the `Patient` defined at `5-1-5:15`. This can be shown in this xml: 
```xml
 <statements>
      <def locator="10:1-11:5" resultTypeName="qdm53:Patient" name="Patient" context="Patient">
         <expression xsi:type="SingletonFrom">
            <operand locator="5:1-5:15" dataType="qdm53:Patient" templateId="Patient" xsi:type="Retrieve"/>
         </expression>
      </def>
      <def locator="7:1-8:10" resultTypeName="qdm53:Patient" name="Initial Population" context="Patient" accessLevel="Public">
         <expression locator="8:2-8:10" resultTypeName="qdm53:Patient" name="Patient" xsi:type="ExpressionRef"/>
      </def>
   </statements>
```

When using this CQL: 
```
library test version '0.0.000'

using QDM version '5.3'

context Population

define "Initial Population":
	"Patient"

define "Patient":
	true
```

In this CQL, the definition `Patient` has a result type of `QDM.Patient`. It is also referencing the expression from line `5:1-5-18` however, this line is talking about `Population` and not `Patient`. This expression should return a `System.Boolean`. Also, the definition `Initial Population` has a result type of `QDM.Patient` and not `System.Boolean`. This can be shown with this ELM output: 

```xml
   <statements>
      <def locator="10:1-11:5" resultTypeName="qdm53:Patient" name="Patient" context="Population">
         <expression xsi:type="SingletonFrom">
            <operand locator="5:1-5:18" dataType="qdm53:Patient" templateId="Patient" xsi:type="Retrieve"/>
         </expression>
      </def>
      <def locator="7:1-8:10" resultTypeName="qdm53:Patient" name="Initial Population" context="Population" accessLevel="Public">
         <expression locator="8:2-8:10" resultTypeName="qdm53:Patient" name="Patient" xsi:type="ExpressionRef"/>
      </def>
   </statements>
```

When using this CQL:
```
library test version '0.0.000'

using QDM version '5.3'

context Population

define "Initial Population":
	true
```

`Patient` is still located in the statements as an expression definition. This can be seen through this ELM output: 
```xml
      <def locator="5:1-5:18" name="Patient" context="Population">
         <expression xsi:type="SingletonFrom">
            <operand locator="5:1-5:18" dataType="qdm53:Patient" templateId="Patient" xsi:type="Retrieve"/>
         </expression>
      </def>
      <def locator="7:1-8:8" resultTypeName="qdm53:Patient" name="Initial Population" context="Population" accessLevel="Public">
         <expression locator="8:2-8:8" resultTypeName="qdm53:Patient" name="Patient" xsi:type="ExpressionRef"/>
      </def>
   </statements>
```

Should `Patient` always be coming out in the list of statements? What is the intended behavior supposed to be when their is a definition named `Patient`?

**Comment by brynrhodes** [comment]
One of the things that the model info specifies is what "Patient" means, so that the `context Patient` statement has meaning. Whenever a `context` statement is encountered, the translator will automatically add the `Patient` definition to the library, referencing the patient class specified by the model info.

So the translator should be complaining about a duplicate identifier when the CQL itself also contains a definition for `Patient`, but that isn't happening. So I'm flagging this as a defect.

As far as `Patient` versus `Population` context, the `Patient` definition is always `Patient` context, so you can reference it from a `Population` context expression, but if you wanted a `Population` context that was the list of all patients, you would have to define a new expression for that, something like:

    context Population
    define Patients: Patient

When you reference a `Patient` context expression from a `Population` context, the result is a flattened list of the `Patient` context expression evaluated for each element of the `Population`.

Note that the current CQL-Based HQMF IG requires that measures use only `Patient` context expressions, so you shouldn't encounter any measures that use `Population` context yet. We are considering a question in the current ballot about whether that restriction should be relaxed, so if you have any input on that, we'd be interested in your thoughts.

**Comment by brynrhodes** [comment]
Included in the [1.3 CQL Ballot](https://github.com/cqframework/clinical_quality_language/releases/tag/v1.3.0-SNAPSHOT) release.
